src/order/hom/basic.lean

```diff
@@ -676,27 +676,44 @@ protected noncomputable def strict_mono_on.order_iso {α β} [linear_order α] [
 { to_equiv := hf.inj_on.bij_on_image.equiv _,
   map_rel_iff' := λ x y, hf.le_iff_le x.2 y.2 }
 
+namespace strict_mono
+
+variables {α β} [linear_order α] [preorder β]
+variables (f : α → β) (h_mono : strict_mono f) (h_surj : function.surjective f)
+
 /-- A strictly monotone function from a linear order is an order isomorphism between its domain and
 its range. -/
-protected noncomputable def strict_mono.order_iso {α β} [linear_order α] [preorder β] (f : α → β)
-  (h_mono : strict_mono f) : α ≃o set.range f :=
+@[simps] protected noncomputable def order_iso : α ≃o set.range f :=
```

Comment:
Do you maybe want 
```suggestion
@[simps apply] protected noncomputable def order_iso : α ≃o set.range f :=
```
here? Otherwise, it's not clear to me that you need `order_iso_of_surjective_symm_apply_self` below, as the `symm` direction will always be unfolded by the lemma here first.

Reply:
OK let's go with this.